src/runtime/converter.cs

```diff
@@ -859,7 +856,22 @@ private static bool ToArray(IntPtr value, Type obType, out object result, bool s
             result = null;
 
             bool IsSeqObj = Runtime.PySequence_Check(value);
-            var len = IsSeqObj ? Runtime.PySequence_Size(value) : -1;
+            int len = -1;
+            if (IsSeqObj)
+            {
+                long longLen = Runtime.PySequence_Size(value);
+                if (longLen > int.MaxValue)
+                {
+                    if (setError)
+                    {
+                        Exceptions.SetError(Exceptions.NotImplementedError, ".NET does not support lists with over 2^31 elements");
+                    }
+
+                    return false;
+                }
+
+                len = longLen < 0 ? -1 : checked((int)longLen);
+            }
```

Comment:
This still fails to finish on Windows 32bit right now, which hints a problem with the way you handle the result of `Runtime.PySequence_Size`. As far as I can tell, the changes you did there are irrelevant for the actual problem you wanted to fix, could you please revert these changes as well as the one below and see whether it passes then?

Reply:
They are sort of relevant. `KeyError` from the test I gets `true` on `PySequence_Check`. I will rewrite the test, and move the fix into a separate PR for clarity.